### ISSUE TITLE
[bug][android] duplicated source requests sent in looped videos?

### ISSUE THREAD

**Issue by lovegaoshi** [TheWidlarzGroup/react-native-video#3428]
# Bug

## Platform
Which player are you experiencing the problem on:
* Android

## Environment info

Library version: 6.0.0 beta2
Device: android 14 samsung S21, emulator

## Steps To Reproduce
I don't see this behavior in 5.2.1, started to see this in v6; expo-av when ejected to bare workflow also has the same behavior if it's any relevant, but not when managed (??)

1. set a video to loop
2. monitor httpToolkit
3. the source (*.bilivideo.com) is periodically requested
![Screenshot from 2023-12-18 12-43-32](https://github.com/react-native-video/react-native-video/assets/106490582/0ad0eb80-3dc3-408d-8cfc-3f58b4b77865)

...

## Expected behaviour

the source is only requested once

## Reproducible sample code
https://github.com/lovegaoshi/videotest/tree/rnv6

## Video sample
If possible, include a link to the video that has the problem that can be streamed or downloaded from.

**Comment by lovegaoshi** [comment]
after some initial investigations:

[this issue ](https://github.com/google/ExoPlayer/issues/8038) was describing the exact same behavior, however it was due to using `exoplayer.setMediaItem` instead of` setMediaSource`, when migrating away from the depreciated `exoplayer.prepare(mediaSource)`. while rnv5.2.1 does use `exoplayer.prepare`, rnv6 also does use `setMediaSource`. using exoplayer.prepare in rnv6 doesnt magically fix the issue either.

then I'm considering this to be a problem with the media3 migration but the [google media3 demo repo](https://github.com/androidx/media) uses `exoplayer.setMediaItems` and I suspect this might be circling back to the problem described in the above issue; both media3 and exoplayer2 keep requesting the source as below:
![Screenshot from 2024-01-03 09-52-21](https://github.com/react-native-video/react-native-video/assets/106490582/c149741b-a1e6-4ef0-9b8d-0f1611042bf5)
I also tried to [shove a simpleCache](https://github.com/lovegaoshi/react-native-video/commit/8346be63548d939015adf0bcf1d023346821c4fa) in RNV following [this](https://exoplayer.dev/customization.html#caching-data-loaded-from-the-network) but while it did create the cache folder, nothing is actually stored within.

next i'll try to use setMediaSource in media3 and if this still occurs, I'll raise the issue to google; otherwise something was wrong in the RNV repo.

have no idea why expo has that weird behavior.

**Comment by lovegaoshi** [comment]
I got some [clarifications from the media3 team](https://github.com/androidx/media/issues/979) and was able to make a caching system work; so not directly fixing this issue (bug?) but caching will effectively stop the repeated requests as well. I saw in the docs RNV only supports caching for iOS and from a quick repo search I dont see SimpleCache; is it still correct RNV doesnt have cache for android? I can implement that and migrate my app to RNV6 then.
